### PR TITLE
GetNumPhysicalCPUs() without using libcpuid

### PR DESCRIPTION
**Pull request by rokups**

### CODE REVIEW

File: Source/Urho3D/Core/ProcessUtils.cpp [hunk 1]
```diff
@@ -412,6 +473,61 @@ ea::string GetPlatformName()
...
+    return 1; // Targeting a single-threaded Emscripten build.
+#endif
+#elif defined(_WIN32)
+    unsigned cpuCount = 0;
+    DWORD bufferSize = 0;
+    GetLogicalProcessorInformation(NULL, &bufferSize);
+    if (ERROR_INSUFFICIENT_BUFFER == GetLastError())
+    {
+        const size_t entryCount = bufferSize / sizeof(SYSTEM_LOGICAL_PROCESSOR_INFORMATION);
+        std::vector<SYSTEM_LOGICAL_PROCESSOR_INFORMATION> entries(entryCount);
```
Comment (iSLC) [inline Source/Urho3D/Core/ProcessUtils.cpp:515]: `std::vector`?

**Comment by rokups** [comment]
I guess we could optimize it with some fixed container from eastl yeah

File: Source/Urho3D/Core/ProcessUtils.cpp [hunk 2]
```diff
@@ -412,6 +423,63 @@ ea::string GetPlatformName()
...
+        while (fgets(line, sizeof(line), fp))
+        {
+            int id = 0;
+            if (sscanf(line, "core id%*[\t]: %d", &id) == 1)
+                cpuCount = Max(cpuCount, static_cast<unsigned>(id + 1));
+        }
+        fclose(fp);
+    }
+#elif defined(__EMSCRIPTEN__)
+#ifdef __EMSCRIPTEN_PTHREADS__
```
Comment (eugeneko) [inline Source/Urho3D/Core/ProcessUtils.cpp:454]: How is that better than `std::thread::hardware_concurrency`?
Also, let's use macros from here whenever possible (I don't know if it's possible in this specific case):
https://github.com/rbfx/rbfx/blob/master/Source/Urho3D/Urho3D.h#L113

File: Source/Urho3D/Core/ProcessUtils.cpp [hunk 3]
```diff
@@ -412,6 +423,63 @@ ea::string GetPlatformName()
...
+    cpuCount = emscripten_num_logical_cores();
+#else
+    cpuCount = 1; // Targeting a single-threaded Emscripten build.
+#endif  // __EMSCRIPTEN_PTHREADS__
+#elif defined(_WIN32)
```
Comment (eugeneko) [inline Source/Urho3D/Core/ProcessUtils.cpp:459]: Let's use enki::GetNumHardwareThreads for Windows.
Also, we can remove platforms if they add more platforms in their library.

File: Source/Urho3D/Core/ProcessUtils.cpp (hunk 2 above)
Comment (rokups) [reply Source/Urho3D/Core/ProcessUtils.cpp:454]: Right, its not better. I just restored old code 😁

File: Source/Urho3D/Core/ProcessUtils.cpp (hunk 3 above)
Comment (rokups) [reply Source/Urho3D/Core/ProcessUtils.cpp:459]: `enki::GetNumHardwareThreads` returns number of logical cores, not fit for the purpose

Comment (eugeneko) [reply Source/Urho3D/Core/ProcessUtils.cpp]: Right, I missed it, thanks.
FYI, a bit of context: https://github.com/dougbinks/enkiTS/issues/97

Comment (rokups) [reply Source/Urho3D/Core/ProcessUtils.cpp]: Uh oh, given that i made our `GetNumLogicalCPUs()` use that as it is more correct.

**Comment by rokups** [comment]
@eugeneko, approval? 😁

**Review by eugeneko — APPROVED** [review]
As long as it's not me maintaining it

**Pull request merged**